utilities/utils/utils.go

```diff
@@ -400,7 +400,14 @@ func AddContainer(state string, container types.Container, containers []model.Pi
 		Labels:   container.Labels,
 		Created:  container.Created,
 	}
-	return append(containers, containerData)
+	if state == "stopped" {
+		inspect, err := dockerClient.ContainerInspect(context.Background(), container.ID)
```

Comment:
Now that I see it, introducing an inspect call in the ping lgoci would be way too expensive we cannot implement it this way.

Here's how it should be implemented:
When the agent does a stop (compute.instance.deactivate) -real or no-op-, it should do the inspect at that point and return the exit code as part of that response. In fact, you could probably just add exitCode to the `getInstanceHostMapData` function.

Reply:
👍